core/src/main/java/jenkins/util/AntWithFindResourceClassLoader.java

```diff
@@ -0,0 +1,60 @@
+package jenkins.util;
+
+import org.apache.tools.ant.Project;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.net.URL;
+import java.util.Collection;
+import java.util.Enumeration;
+import java.util.Vector;
+
+/**
+ * As of 1.8.0, {@link org.apache.tools.ant.AntClassLoader} doesn't implement {@link #findResource(String)}
+ * in any meaningful way, which breaks fast lookup. Implement it properly.
+ */
+public class AntWithFindResourceClassLoader extends AntClassLoader implements Closeable {
```

Comment:
Prefer `@Restricted(NoExternalUse.class)`. Or just move into `hudson` package and leave it as default access.